### PR TITLE
Fix MacOS 10.14 nothing rendering until resize

### LINKED ISSUE #972: Amethyst examples don't render anything on mac until window is resized
When trying to run any of the examples, a blank window appears until I manually resize it. For examples that have sound or other forms of output it still works in the background so I know the engine is running but rendering doesn't work. 

Computer is a 13 inch macbook pro running mojave with intel 655 gpu

Stdout from the Sprites example follows:

 cargo run --example sprites
    Finished dev [unoptimized + debuginfo] target(s) in 0.36s                                                                                                              
     Running `target/debug/examples/sprites`
[INFO][gfx_device_gl::info] parsing version '4.1 INTEL-12.0.34'
[INFO][gfx_device_gl::info] parsing version '4.10'
[INFO][gfx_device_gl] Info { platform_name: PlatformName { vendor: "Intel Inc.", renderer: "Intel(R) Iris(TM) Plus Graphics 655" }, version: 4.1, INTEL-12.0.34, shading_language: 4.10, extensions: {"GL_ARB_internalformat_query", "GL_ARB_draw_buffers_blend", "GL_ARB_texture_gather", "GL_NV_texture_barrier", "GL_APPLE_texture_range", "GL_ARB_draw_indirect", "GL_EXT_debug_marker", "GL_ARB_viewport_array", "GL_ARB_sampler_objects", "GL_ARB_texture_storage", "GL_ARB_texture_swizzle", "GL_ARB_vertex_attrib_64bit", "GL_ARB_gpu_shader5", "GL_ARB_texture_cube_map_array", "GL_ARB_texture_buffer_object_rgb32", "GL_ARB_blend_func_extended", "GL_ARB_ES2_compatibility", "GL_ARB_separate_shader_objects", "GL_ARB_timer_query", "GL_EXT_texture_sRGB_decode", "GL_APPLE_client_storage", "GL_APPLE_row_bytes", "GL_EXT_debug_label", "GL_ARB_transform_feedback2", "GL_ARB_transform_feedback3", "GL_EXT_framebuffer_multisample_blit_scaled", "GL_EXT_texture_compression_s3tc", "GL_ARB_shading_language_include", "GL_ARB_tessellation_shader", "GL_ARB_shader_subroutine", "GL_ARB_texture_rgb10_a2ui", "GL_ARB_occlusion_query2", "GL_APPLE_rgb_422", "GL_ATI_texture_mirror_once", "GL_APPLE_flush_render", "GL_ARB_instanced_arrays", "GL_ARB_vertex_type_2_10_10_10_rev", "GL_APPLE_container_object_shareable", "GL_ARB_texture_query_lod", "GL_ARB_sample_shading", "GL_ARB_gpu_shader_fp64", "GL_ARB_shader_bit_encoding", "GL_EXT_texture_filter_anisotropic", "GL_ARB_explicit_attrib_location", "GL_APPLE_object_purgeable"} }
[INFO][gfx_device_gl] Capabilities { max_vertex_count: 1048575, max_index_count: 150000, max_texture_size: 16384, max_patch_size: 32, instance_base_supported: false, instance_call_supported: true, instance_rate_supported: true, vertex_base_supported: true, srgb_color_supported: true, constant_buffer_supported: true, unordered_access_view_supported: true, separate_blending_slots_supported: true, copy_buffer_supported: true }
[INFO][gfx_device_gl] PrivateCaps { array_buffer_supported: true, frame_buffer_supported: true, immutable_storage_supported: true, sampler_objects_supported: true, program_interface_supported: false, buffer_storage_supported: false, clear_buffer_supported: true, frag_data_location_supported: true, sampler_lod_bias_supported: true, texture_border_clamp_supported: true }
[INFO][gfx_device_gl::factory]  Created frame buffer 1
[DEBUG][amethyst_renderer::pipe::effect] Building effect
[DEBUG][amethyst_renderer::pipe::effect] Compiling shaders
[INFO][gfx_device_gl::shade]    Compiled shader 1
[INFO][gfx_device_gl::shade]    Compiled shader 2
[DEBUG][amethyst_renderer::pipe::effect] Creating pipeline state
[INFO][gfx_device_gl::shade]    Linked program 3
[WARN][gfx_device_gl::shade]    Log: WARNING: Could not find fragment shader output 'Target2' to match FragDataBinding request.
WARNING: Could not find fragment shader output 'Target3' to match FragDataBinding request.
WARNING: Could not find fragment shader output 'Target1' to match FragDataBinding request.
WARNING: Could not find fragment shader output 'Target0' to match FragDataBinding request.

[INFO][gfx_device_gl::shade]            Sampler[0] = 'albedo'   F32     D2(NoArray, NoMultiSample)
[INFO][gfx_device_gl::shade]            Attrib[0] = "depth"     F32     Single
[INFO][gfx_device_gl::shade]            Attrib[6] = "gl_VertexID"       I32     Single
[INFO][gfx_device_gl::shade]            Attrib[1] = "pos"       F32     Vector(2)
[INFO][gfx_device_gl::shade]            Attrib[2] = "v_offset"  F32     Vector(2)
[INFO][gfx_device_gl::shade]            Attrib[3] = "u_offset"  F32     Vector(2)
[INFO][gfx_device_gl::shade]            Attrib[4] = "dir_y"     F32     Vector(2)
[INFO][gfx_device_gl::shade]            Attrib[5] = "dir_x"     F32     Vector(2)
[INFO][gfx_device_gl::shade]            Block[0] = 'ViewArgs' of size 128
[INFO][gfx_device_gl::shade]                    Element at 0    = 'proj'        F32     Matrix(ColumnMajor, 4, 4)
[INFO][gfx_device_gl::shade]                    Element at 64   = 'view'        F32     Matrix(ColumnMajor, 4, 4)
[DEBUG][gfx_device_gl::shade] Program 3 reflection: ProgramInfo { vertex_attributes: [AttributeVar { name: "depth", slot: 0, base_type: F32, container: Single }, AttributeVar { name: "pos", slot: 1, base_type: F32, container: Vector(2) }, AttributeVar { name: "v_offset", slot: 2, base_type: F32, container: Vector(2) }, AttributeVar { name: "u_offset", slot: 3, base_type: F32, container: Vector(2) }, AttributeVar { name: "dir_y", slot: 4, base_type: F32, container: Vector(2) }, AttributeVar { name: "dir_x", slot: 5, base_type: F32, container: Vector(2) }], globals: [], constant_buffers: [ConstantBufferVar { name: "ViewArgs", slot: 0, size: 128, usage: VERTEX, elements: [ConstVar { name: "proj", location: 0, count: 1, base_type: F32, container: Matrix(ColumnMajor, 4, 4) }, ConstVar { name: "view", location: 64, count: 1, base_type: F32, container: Matrix(ColumnMajor, 4, 4) }] }], textures: [TextureVar { name: "albedo", slot: 0, base_type: F32, ty: D2(NoArray, NoMultiSample), usage: VERTEX | PIXEL }], unordereds: [], samplers: [SamplerVar { name: "albedo", slot: 0, ty: SamplerType(NoCompare, NoRect), usage: VERTEX | PIXEL }], outputs: [], output_depth: false, knows_outputs: false }
[DEBUG][amethyst_renderer::pipe::effect] Creating raw constant buffers
[INFO][gfx_device_gl::factory]  Created buffer 1
[INFO][gfx_device_gl::factory]  Created buffer 2
[DEBUG][amethyst_renderer::pipe::effect] Set global uniforms
[DEBUG][amethyst_renderer::pipe::effect] Process Color/Depth/Blend outputs
[DEBUG][amethyst_renderer::pipe::effect] Finished building effect
[INFO][amethyst::app] Initializing Amethyst...
[INFO][amethyst::app] Version: 0.8.0
[INFO][amethyst::app] Platform: x86_64-apple-darwin
[INFO][amethyst::app] Amethyst git commit: 1a3d9a5a8e143be815d64a9cd07f3e407e9e952d
[INFO][amethyst::app] Rustc version: 1.30.0-nightly Nightly
[INFO][amethyst::app] Rustc git commit: 5c875d93855c6d577962b0f74f17374f37b219c9
[DEBUG][amethyst_assets::loader] "renderer::Texture": Loading asset "texture/bat.32x32.png" with format "PNG" from source "[default source]" (handle id: Handle { id: 7, marker: PhantomData })
[DEBUG][sprites::sprite_sheet_loader] 0: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.0, right: 0.16666667, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 1: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.16666667, right: 0.33333334, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 2: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.33333334, right: 0.5, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 3: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.5, right: 0.6666667, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 4: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.6666667, right: 0.8333333, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 5: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.8333333, right: 1.0, bottom: 0.5, top: 1.0 } }
[DEBUG][sprites::sprite_sheet_loader] 6: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.0, right: 0.16666667, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites::sprite_sheet_loader] 7: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.16666667, right: 0.33333334, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites::sprite_sheet_loader] 8: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.33333334, right: 0.5, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites::sprite_sheet_loader] 9: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.5, right: 0.6666667, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites::sprite_sheet_loader] 10: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.6666667, right: 0.8333333, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites::sprite_sheet_loader] 11: Sprite: Sprite { width: 32.0, height: 32.0, offsets: [16.0, 16.0], tex_coords: TextureCoordinates { left: 0.8333333, right: 1.0, bottom: 0.0, top: 0.5 } }
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 0, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 1, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 2, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 3, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 4, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 5, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 6, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 7, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 8, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 9, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 10, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 11, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 0, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 1, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 2, flip_horizontal: false, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 3, flip_horizontal: true, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 4, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 5, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 6, flip_horizontal: false, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 7, flip_horizontal: true, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 8, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 9, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 10, flip_horizontal: false, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 11, flip_horizontal: true, flip_vertical: true }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 0, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 1, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 2, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 3, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 4, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 5, flip_horizontal: false, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 6, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 7, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 8, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 9, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 10, flip_horizontal: true, flip_vertical: false }`
[DEBUG][sprites] sprite_render: `SpriteRender { sprite_sheet: Handle { id: 0, marker: PhantomData }, sprite_number: 11, flip_horizontal: true, flip_vertical: false }`
[DEBUG][amethyst_assets::storage] "animation::Animation": Asset "<Data>" (handle id: Handle { id: 0, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::SpriteSheet": Asset "<Data>" (handle id: Handle { id: 0, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "animation::Sampler": Asset "<Data>" (handle id: Handle { id: 0, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "animation::Animation": Asset "<Data>" (handle id: Handle { id: 1, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "animation::Sampler": Asset "<Data>" (handle id: Handle { id: 1, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "animation::Sampler": Asset "<Data>" (handle id: Handle { id: 2, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "animation::Sampler": Asset "<Data>" (handle id: Handle { id: 3, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 0, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 1, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 2, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 3, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 4, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 5, marker: PhantomData }) has been loaded successfully
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "<Data>" (handle id: Handle { id: 6, marker: PhantomData }) has been loaded successfully
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[WARN][amethyst_renderer::pass::sprite::interleaved] Texture not loaded for texture id: `0`.
[DEBUG][amethyst_assets::storage] "renderer::Texture": Asset "texture/bat.32x32.png" (handle id: Handle { id: 7, marker: PhantomData }) has been loaded successfully
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][gfx_device_gl::factory]  Created buffer 3
[INFO][amethyst::app] Engine is shutting down

### PR DESCRIPTION
**Pull request by LucioFranco**
This fix adds window resize to happen after window creation on macos
and opengl. This is a work around proposed in #972. This fix should be
temporary until there is a fix upstream. This shouldn't affect users
as the resize happens only after window creation and only then. This
closes #972.

### CODE REVIEW

File: amethyst_renderer/src/renderer.rs [hunk 1]
```diff
@@ -316,3 +316,41 @@ fn init_backend(wb: WindowBuilder, el: &EventsLoop, config: &DisplayConfig) -> R
...
+        DepthBuffer {
+            as_input: None,
+            as_output: depth,
+        },
+        size,
+    );
+
+    // On Mac 10.14 we need to resize the window after creation
+    // this is related to this issue amethyst/amethyst#972
+    el.poll_events(|_| {});
```
Comment (Moxinilian) [inline amethyst_renderer/src/renderer.rs]: Maybe we could annotate this more so that we don't forget to remove it when the bug is fixed?

Comment (LucioFranco) [reply amethyst_renderer/src/renderer.rs]: @Moxinilian does a TODO work or a follow-up issue?

Comment (Moxinilian) [reply amethyst_renderer/src/renderer.rs]: I think an issue might be better, but at the same time if it ends up at the bottom of the pile...

(I wonder if we have old critical issues we forgot about)

Comment (LucioFranco) [reply amethyst_renderer/src/renderer.rs]: Ive added a fixme and ill create an issue here shortly.

File: amethyst_renderer/src/renderer.rs [hunk 2]
```diff
@@ -316,3 +316,42 @@ fn init_backend(wb: WindowBuilder, el: &EventsLoop, config: &DisplayConfig) -> R
...
+
+    // FIXME: Resolve #972 and remove this extra resize
+    // On Mac 10.14 we need to resize the window after creation
+    // this is related to this issue amethyst/amethyst#972
+    el.poll_events(|_| {});
+    let (width, height): (u32, u32) = win
+        .get_outer_size()
+        .expect("Window no longer exists")
+        .into();
+    win.resize((width, height).into());
```
[the remark this answers is not in the document]
Comment (Rhuagh) [reply amethyst_renderer/src/renderer.rs]: Yes

[the remark this answers is not in the document]
Comment (torkleyy) [reply amethyst_renderer/src/renderer.rs]: Yeah saw that afterwards, marking as resolved.

**Comment by LucioFranco** [comment]
bors r=Moxinilian, Torkleyy, azriel91

**Pull request merged**